server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -293,6 +292,11 @@
     }, Setting.Property.NodeScope);
 
     private static final String CLIENT_TYPE = "node";
+    public static final Setting<TimeValue> INITIAL_STATE_TIMEOUT_SETTING = Setting.positiveTimeSetting(
```

Comment:
That looks a bit off: why discovery setting on node level? May be it is better keep old `DiscoverySettings` class with this single setting (for now)?

Reply:
It's a `NodeScope` setting refactored from `DiscoverySettings` and the only remaining setting for node discovery so I think it's appropriately refactored here to node level without the overhead of an inner class?

Reply:
Sorry, should have been more clear, `NodeScope` is totally fine, I think the pretty useful convention we have is that that `Node` settings are prefixed with `node.`, `Search` settings with `search.`, `Index` settings with `index.`, the overhead of inner class is super minimal but it would be just natural to look for `discovery.` under `Discovery` - related classes, just my two cents.

Reply:
oic what you're saying. API wise? sure we can make a public inner class. And I think "overhead" should be 1:1? Instead of loading `DiscoverySettings.class` (which is now removed) we're adding a new `Node$DiscoverySettings.class`.

Reply:
:+1:  I refactored `DiscoverySettings` as an internal class to `Node`. I think that's a nice clean suggestion.

Reply:
Thank you @nknize !